tools/slicec-cs/src/decoding.rs

```diff
@@ -301,11 +316,22 @@ decoder.DecodeSequenceOfOptionals(
         )
     } else {
         match element_type.concrete_type() {
-            Types::Primitive(primitive) if primitive.fixed_wire_size().is_some() => {
+            Types::Primitive(primitive) if primitive.is_numeric_or_bool() && primitive.fixed_wire_size().is_some() => {
```

Comment:
This extra `is_numeric_or_bool` check is redundant.
We could probably remove it from the other half of this massive if/else.

All primitives with fixed wire size are either numeric or a bool.
And for decoding, what we really care about is that's fixed size, not so much the actual type.